actionview/lib/action_view/helpers/form_helper.rb

```diff
@@ -1074,68 +1074,37 @@ def time_field(object_name, method, options = {})
-      #   # => <input id="user_born_on" name="user[born_on]" type="datetime" value="1984-01-12T00:00:00.000+0000" />
-      #
-      # You can create values for the "min" and "max" attributes by passing
-      # instances of Date or Time to the options hash.
-      #
-      #   datetime_field("user", "born_on", min: Date.today)
-      #   # => <input id="user_born_on" name="user[born_on]" type="datetime" min="2014-05-20T00:00:00.000+0000" />
-      #
-      # Alternatively, you can pass a String formatted as an ISO8601 datetime
-      # with UTC offset as the values for "min" and "max."
-      #
-      #   datetime_field("user", "born_on", min: "2014-05-20T00:00:00+0000")
-      #   # => <input id="user_born_on" name="user[born_on]" type="datetime" min="2014-05-20T00:00:00.000+0000" />
-      #
-      def datetime_field(object_name, method, options = {})
-        ActiveSupport::Deprecation.warn(<<-MESSAGE.squish)
-          datetime_field is deprecated and will be removed in Rails 5.1.
-          Use datetime_local_field instead.
-        MESSAGE
-        Tags::DatetimeField.new(object_name, method, self, options).render
-      end
-
       # Returns a text_field of type "datetime-local".
       #
-      #   datetime_local_field("user", "born_on")
+      #   datetime_field("user", "born_on")
       #   # => <input id="user_born_on" name="user[born_on]" type="datetime-local" />
       #
       # The default value is generated by trying to call +strftime+ with "%Y-%m-%dT%T"
       # on the object's value, which makes it behave as expected for instances
       # of DateTime and ActiveSupport::TimeWithZone.
       #
       #   @user.born_on = Date.new(1984, 1, 12)
-      #   datetime_local_field("user", "born_on")
+      #   datetime_field("user", "born_on")
       #   # => <input id="user_born_on" name="user[born_on]" type="datetime-local" value="1984-01-12T00:00:00" />
       #
       # You can create values for the "min" and "max" attributes by passing
       # instances of Date or Time to the options hash.
       #
-      #   datetime_local_field("user", "born_on", min: Date.today)
+      #   datetime_field("user", "born_on", min: Date.today)
       #   # => <input id="user_born_on" name="user[born_on]" type="datetime-local" min="2014-05-20T00:00:00.000" />
       #
       # Alternatively, you can pass a String formatted as an ISO8601 datetime as
       # the values for "min" and "max."
       #
-      #   datetime_local_field("user", "born_on", min: "2014-05-20T00:00:00")
+      #   datetime_field("user", "born_on", min: "2014-05-20T00:00:00")
       #   # => <input id="user_born_on" name="user[born_on]" type="datetime-local" min="2014-05-20T00:00:00.000" />
       #
-      def datetime_local_field(object_name, method, options = {})
+      def datetime_field(object_name, method, options = {})
         Tags::DatetimeLocalField.new(object_name, method, self, options).render
       end
 
+      alias datetime_local_field datetime_field
```

Comment:
Do we need to change the deprecation shipping with 5.0 if we are not deprecating this, and `alias`ing instead? 😬